src/components/themed/PinDots.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import * as React from 'react'
+import { View } from 'react-native'
+
+import { type Theme, cacheStyles, useTheme } from '../services/ThemeContext.js'
+
+export function PinDots(props: { pin: string }) {
+  const theme = useTheme()
+  const styles = getStyles(theme)
+  return (
+    <View style={styles.container}>
+      <View style={[styles.circle, props.pin.length > 0 && styles.circleFilled]} />
```

Comment:
Make this part a bit more 'generic'.
Use a map/loop instead of manually adding and checking each circle.